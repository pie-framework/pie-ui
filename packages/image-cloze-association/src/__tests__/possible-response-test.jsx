import React from 'react';
import _ from 'lodash';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import PossibleResponse from '../possible-response';

describe('Possible Response', () => {
  let wrapper;

  const mkWrapper = (opts = {}) => {
    opts = _.extend(
      {
        canDrag: false,
        containerStyle: {},
        data: {}
      },
      opts
    );

    return shallow(<PossibleResponse {...opts} />);
  };

  beforeEach(() => {
    wrapper = mkWrapper();
  });

  describe('snapshots', () => {
    describe('canDrag', () => {
      it('renders', () => {
        const wrapper = mkWrapper({ canDrag: false });
        expect(toJson(wrapper)).toMatchSnapshot();
      });
    });

    describe('data', () => {
      it('renders', () => {
        const wrapper = mkWrapper({ data: { id: 1, value: '1', containerIndex: 0 } });
        expect(toJson(wrapper)).toMatchSnapshot();
      });
    });
  });
});
