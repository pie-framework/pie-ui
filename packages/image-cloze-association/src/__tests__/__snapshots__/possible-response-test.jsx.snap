// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Possible Response snapshots canDrag renders 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`Possible Response snapshots data renders 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
